File: common/py/llamaindex/openai.py
from agentuity import AgentRequest, AgentResponse, AgentContext
from llama_index.llms.openai import OpenAI
from llama_index.core.llms import ChatMessage
import os

# TODO: Add your key via `agentuity env set --secret OPENAI_API_KEY`
# Get your API key here: https://platform.openai.com/settings/organization/api-keys
api_key = os.getenv("OPENAI_API_KEY")

if not api_key:
    raise ValueError("OPENAI_API_KEY environment variable not set.")

client = OpenAI(
    model="gpt-5-mini",
    api_key=api_key
)

def welcome():
    return {
        "welcome": "Welcome to the LlamaIndex Agent with OpenAI! I can help you build AI-powered applications using LlamaIndex and OpenAI models.",
        "prompts": [
            {
                "data": "How do I use LlamaIndex to call OpenAI models?",
                "contentType": "text/plain"
            },
            {
                "data": "What are the best practices for prompt engineering with LlamaIndex?",
                "contentType": "text/plain"
            }
        ]
    }

async def run(request: AgentRequest, response: AgentResponse, context: AgentContext):
    try:
        result = client.chat([
            ChatMessage(
                role="system",
                content="You are a helpful assistant that provides concise and accurate information."
            ),
            ChatMessage(
                role="user",
                content=await request.data.text() or "Hello, OpenAI"
            )
        ])

        return response.text(str(result))
    except Exception as e:
        context.logger.error(f"Error running agent: {e}")

        return response.text("Sorry, there was an error processing your request.")

File: common/js/vercel/openai.ts
import type { AgentRequest, AgentResponse, AgentContext } from "@agentuity/sdk";
import { generateText } from "ai";
import { openai } from "@ai-sdk/openai";

export const welcome = () => {  
	return {  
	  welcome: "Welcome to the Vercel AI SDK with OpenAI Agent! I can help you build AI-powered applications using Vercel's AI SDK with OpenAI models.",  
	  prompts: [  
		{  
		  data: "How do I implement streaming responses with the Vercel AI SDK?",  
		  contentType: "text/plain",  
		},  
		{  
		  data: "Generate a poem about artificial intelligence",  
		  contentType: "text/plain",  
		}  
	  ],  
	};  
  };

export default async function Agent(
	req: AgentRequest,
	resp: AgentResponse,
	ctx: AgentContext,
) {
	const res = await generateText({
		model: openai("gpt-4o"),
		system: "You are a friendly assistant!",
		prompt: (await req.data.text()) ?? "Why is the sky blue?",
	});
	return resp.text(res.text);
}
